src/tbtcv2-rewards/rewards.ts

```diff
@@ -0,0 +1,726 @@
+  REQUIRED_MIN_PRE_PARAMS,
+  ALLOWED_UPGRADE_DELAY,
+  PRECISION,
+  QUERY_STEP,
+  QUERY_RESOLUTION,
+  HUNDRED,
+  APR,
+} from "./rewards-constants";
+
+program
+  .version("0.0.1")
+  .requiredOption(
+    "-s, --start-timestamp <timestamp>",
+    "starting time for rewards calculation"
+  )
+  .requiredOption(
+    "-e, --end-timestamp <timestamp>",
+    "ending time for rewards calculation"
+  )
+  .requiredOption(
+    "-b, --start-block <timestamp>",
+    "start block for rewards calculation"
+  )
+  .requiredOption(
+    "-z, --end-block <timestamp>",
+    "end block for rewards calculation"
+  )
+  .requiredOption("-a, --api <prometheus api>", "prometheus API")
+  .requiredOption("-j, --job <prometheus job>", "prometheus job")
+  .requiredOption("-s, --october17 <october 17 block>", "october 17 block")
+  .requiredOption(
+    "-r, --releases <client releases in a rewards interval>",
+    "client releases in a rewards interval"
+  )
+  .requiredOption("-n, --network <name>", "network name")
+  .requiredOption("-o, --output <file>", "output JSON file")
+  .parse(process.argv);
+
+// Parse the program options
+const options = program.opts();
+const prometheusJob = options.job;
+const prometheusAPI = options.api;
+const clientReleases = options.releases.split("|"); // sorted from latest to oldest
+const startRewardsTimestamp = parseInt(options.startTimestamp);
+const endRewardsTimestamp = parseInt(options.endTimestamp);
+const startRewardsBlock = parseInt(options.startBlock);
+const endRewardsBlock = parseInt(options.endBlock);
+const october17Block = parseInt(options.october17);
+const rewardsDataOutput = options.output;
+const network = options.network;
+
+const prometheusAPIQuery = `${prometheusAPI}/query`;
+// Go back in time relevant to the current date to get data for the exact
+// rewards interval dates.
+const offset = Math.floor(Date.now() / 1000) - endRewardsTimestamp;
+
+export async function calculateRewards() {
+  if (Date.now() / 1000 < endRewardsTimestamp) {
+    console.log("End time interval must be in the past");
+    return "End time interval must be in the past";
+  }
+
+  const provider = new ethers.providers.EtherscanProvider(
+    network,
+    process.env.ETHERSCAN_TOKEN
+  );
+
+  const monthlyRate = (APR / 12) * HUNDRED; // monthly periodic rate ARP / 12 month
+  const currentBlockNumber = await provider.getBlockNumber();
+  const rewardsInterval = endRewardsTimestamp - startRewardsTimestamp;
+
+  // Query for bootstrap data that has peer instances grouped by operators
+  const queryBootstrapData = `${prometheusAPI}/query_range`;
+  const paramsBootstrapData = {
+    query: `sum by(chain_address)({job='${prometheusJob}'})`,
+    start: startRewardsTimestamp,
+    end: endRewardsTimestamp,
+    step: QUERY_STEP,
+  };
+
+  const bootstrapData = (
+    await queryPrometheus(queryBootstrapData, paramsBootstrapData)
+  ).data.result;
+
+  const operatorsData = new Array();
+  const rewardsData = new Array();
```

Comment:
```suggestion
  const rewardsData: any = {};
```

Reply:
Sure, part of [c78a6b7](https://github.com/threshold-network/merkle-distribution/pull/35/commits/c78a6b730f5fc7045bf5fbbd34b65be39292a8f4)